KeyVault.Acmebot/Providers/GoDaddyProvider.cs

```diff
@@ -0,0 +1,207 @@
+            foreach (var value in values)
+            {
+                entries.Add(new DnsEntry
+                {
+                    Name = relativeRecordName,
+                    Type = "TXT",
+                    TTL = 600,
+                    Data = value
+                });
+            }
+
+            await _client.AddRecordAsync(zone.Id, entries);
+
+        }
+
+        public async Task DeleteTxtRecordAsync(DnsZone zone, string relativeRecordName)
+        {
+            var records = await _client.ListRecordsAsync(zone.Id);
+
+            var recordsToDelete = records.Where(r => r.Name == relativeRecordName && r.Type == "TXT");
+
+            foreach (var record in recordsToDelete)
+            {
+                await _client.DeleteRecordAsync(zone.Id, record);
+            }
+        }
+
+        private class GoDaddyClient
+        {
+            public GoDaddyClient(string apiKey, string secretKey, string uri)
+            {
+                _httpClient = new HttpClient(new ApiKeyHandler(apiKey, secretKey, new HttpClientHandler()))
+                {
+                    BaseAddress = new Uri(uri)
+                };
+
+                _httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            }
+
+            private readonly HttpClient _httpClient;
+
+            public async Task<IReadOnlyList<ZoneDomain>> ListZonesAsync()
+            {
+                var response = await _httpClient.GetAsync("v1/domains?statuses=ACTIVE");
+
+                response.EnsureSuccessStatusCode();
+
+                var domains = await response.Content.ReadAsAsync<List<ZoneDomain>>();
+
+                return domains;
+            }
+
+            public async Task<IReadOnlyList<DnsEntry>> ListRecordsAsync(string zoneId)
+            {
+                var response = await _httpClient.GetAsync($"v1/domains/{zoneId}/records");
+
+                response.EnsureSuccessStatusCode();
+
+                var entries = await response.Content.ReadAsAsync<List<DnsEntry>>();
+
+                return entries;
+            }
+
+            public async Task DeleteRecordAsync(string zoneId, DnsEntry entry)
+            {
+                var response = await _httpClient.SendAsync(new HttpRequestMessage(HttpMethod.Delete, $"v1/domains/{zoneId}/records/{entry.Type}/{entry.Name}"));
+
+                response.EnsureSuccessStatusCode();
+            }
+
+            public async Task AddRecordAsync(string zoneId, List<DnsEntry> entries)
+            {
+                HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Patch, $"v1/domains/{zoneId}/records")
+                {
+                    Content = new StringContent(JsonConvert.SerializeObject(entries, new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore }), Encoding.UTF8, "application/json")
+                };
+
+                var response = await _httpClient.SendAsync(request);
```

Comment:
Use the methods provided in `HttpClient`.

```suggestion
                var response = await _httpClient.PatchAsync(request, content);
```